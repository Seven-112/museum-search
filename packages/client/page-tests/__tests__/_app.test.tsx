import { shallow } from "enzyme";
import { FunctionComponent } from "react";
import { ApolloProvider } from "react-apollo";
import MyApp from "../../pages/_app";

describe("MyApp component", () => {
  it("Renders the App wrapper.", () => {
    const mockPush = jest.fn();

    const mockRouter = {
      asPath: "/example-path?a=b",
      push: mockPush
    } as any;

    const TestComponent: FunctionComponent = () => <div />;

    const wrapper = shallow(
      <MyApp
        Component={TestComponent}
        pageProps={{ exampleProp: "exampleValue" }}
        router={mockRouter}
      />
    );

    // The first browser render should be empty.
    expect(wrapper.html()).toEqual(null);

    expect(mockPush).lastCalledWith("/example-path?a=b");
    // Normally the router would update the app wrapper, but the mock doesn't, so we force a
    // re-render in the test.
    wrapper.instance().forceUpdate();
    wrapper.update();

    // Renders the Apollo provider.
    expect(wrapper.find(ApolloProvider).exists()).toEqual(true);

    const innerComponent = wrapper.find(TestComponent);
    expect(innerComponent.prop("exampleProp")).toEqual("exampleValue");
  });
});
