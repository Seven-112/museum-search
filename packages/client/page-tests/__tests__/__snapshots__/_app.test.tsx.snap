// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApp component Renders with the ApolloProvider.: MyApp shallow render. 1`] = `
<Container>
  <TestComponent
    exampleProp="exampleValue"
  />
</Container>
`;
